web_poet/mixins.py

```diff
@@ -78,26 +79,23 @@ def urljoin(self, url: Union[str, RequestUrl, ResponseUrl]) -> RequestUrl:
         return RequestUrl(urljoin(self._base_url, str(url)))
 
 
-# TODO: when dropping Python 3.7 support,
-# fix untyped ResponseShortcutsMixin.response using typing.Protocol
```

Comment:
I don't see how does this need a protocol.

Reply:
The idea was to fail the type check if ResponseShortcutsMixin is added to a class without response attribute. But no strong opinion on that.